src/js/components/CheckBoxGroup/doc.js

```diff
@@ -0,0 +1,53 @@
+import { describe, PropTypes } from 'react-desc';
+
+import { getAvailableAtBadge } from '../../utils';
+
+export const doc = CheckBoxGroup => {
+  const DocumentedCheckBoxGroup = describe(CheckBoxGroup)
+    .availableAt(getAvailableAtBadge('CheckBoxGroup'))
+    .description('A group of CheckBoxes.')
+    .usage(
+      `import { CheckBoxGroup } from 'grommet';
+        <CheckBoxGroup />`,
+    )
+    .intrinsicElement('div');
+
+  DocumentedCheckBoxGroup.propTypes = {
+    checked: PropTypes.arrayOf(PropTypes.string).description(
+      `An array of the checked options. If options is provided as an object, the
+      checked array will be the strings of the checked option valueKey.`,
+    ),
+    disabled: PropTypes.bool
+      .description(`Disables all options.`)
+      .defaultValue(undefined),
+    labelKey: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.func,
+    ]).description(
+      `When the options array contains objects, this property indicates how
+        to determine the label of each option. If a string is
+        provided, it is used as the key to retrieve each option's label.`,
+    ),
+    name: PropTypes.string.description(
+      `The DOM name attribute value to use for the underlying <input/> 
+      elements. Required when used in the Context of FormField.`,
+    ),
+    onChange: PropTypes.func.description(
+      `Function that will be called when the user clicks on a CheckBox option. 
+      It will be passed a React event object with a value.`,
+    ),
+    options: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.string),
+      PropTypes.arrayOf(PropTypes.shape({})),
+    ]).description(
+      `Options can be either a string or an object of CheckBox props.`,
```

Comment:
Is it CheckBox props or an arbitrary data object? For Select, it is an arbitrary data object. We do not pass that object as props to the underlying options.

Reply:
It made sense to me to make `options` an object of CheckBox props, this way the user can leverage the propagation of props on an individual CheckBox via `options` objects. (See CheckBoxGroup.js line 81)

RadioButtonGroup is performing this type of propagation in a similar fashion for its nested RadioButton(s).
In the case of CheckBoxGroup `options`, I see more potential similarity with RadioButtonGroup rather than with Select, but I might be missing something.

Reply:
So, for RadioButtonGroup, the options are more focused on the underlying RadioButton. Whereas for Select, the options are more focused on the data list of choices.
For example, if I make a REST call to GET /deployment-targets, which returns a JSON array of objects, I can feed that directly to select and give it labelKey and valueKey appropriate to pick the fields of interest. I cannot do this with CheckBoxGroup or RadioButtonGroup. I would have to map() the result to convert it into something appropriate for the UI presentation.
The UI presentation is less important for Select, since there is no input component that each option maps to. For Select, the children render function is where any translation to UI presentation can happen.
Just thinking about these two approaches and thought I write these notes here for memory.